lib/services/resourceManagement/lib/resource/models/genericResource.js

```diff
@@ -131,7 +131,7 @@ class GenericResource extends models['Resource'] {
             required: false,
             serializedName: 'kind',
             constraints: {
-              Pattern: '^[-\w\._,\(\)]+$'
+              Pattern: /^[-\w\._,\(\)]+$/
```

Comment:
LGTM, `new RegExp(/^[-\w\._,\(\)]+$/).test('Microsoft.IoTHub')` returns true, but I don't have the context for the rest of the files.